src/libaktualizr/primary/aktualizr.cc

```diff
@@ -12,38 +12,19 @@
 using std::make_shared;
 using std::shared_ptr;
 
-Aktualizr::Aktualizr(Config &config) : config_(config) {
-  systemSetup();
-  sig_ = make_shared<boost::signals2::signal<void(shared_ptr<event::BaseEvent>)>>();
-  storage_ = INvStorage::newStorage(config_.storage);
-  storage_->importData(config_.import);
-  uptane_client_ = SotaUptaneClient::newDefaultClient(config_, storage_, sig_);
-}
+Aktualizr::Aktualizr(Config &config)
+    : Aktualizr(config, INvStorage::newStorage(config.storage), std::make_shared<HttpClient>()) {}
 
 Aktualizr::Aktualizr(Config &config, std::shared_ptr<INvStorage> storage_in, std::shared_ptr<HttpInterface> http_in)
-    : config_(config) {
-  systemSetup();
-  sig_ = make_shared<boost::signals2::signal<void(shared_ptr<event::BaseEvent>)>>();
-  storage_ = std::move(storage_in);
-  std::shared_ptr<Bootloader> bootloader_in = std::make_shared<Bootloader>(config_.bootloader, *storage_);
-  std::shared_ptr<ReportQueue> report_queue_in = std::make_shared<ReportQueue>(config_, http_in);
-
-  uptane_client_ = std::make_shared<SotaUptaneClient>(config_, storage_, http_in, bootloader_in, report_queue_in, sig_);
-}
-
-void Aktualizr::systemSetup() {
+    : config_(config), sig_(new event::Channel()) {
   if (sodium_init() == -1) {  // Note that sodium_init doesn't require a matching 'sodium_deinit'
     throw std::runtime_error("Unable to initialize libsodium");
   }
 
-  LOG_TRACE << "Seeding random number generator from /dev/urandom...";
-  Timer timer;
-  unsigned int seed;
-  std::ifstream urandom("/dev/urandom", std::ios::in | std::ios::binary);
-  urandom.read(reinterpret_cast<char *>(&seed), sizeof(seed));
-  urandom.close();
-  std::srand(seed);  // seeds pseudo random generator with random number
```

Comment:
Are you sure it's not used by one of our dependencies? Could be worth checking with some strace.

Reply:
I think we care only about libsodium and libopenssl, and they are not using that.

Reply:
It looks like we originally added randomness initialization for the selection of pretty names in https://github.com/advancedtelematic/aktualizr/commit/6e32f371fa34c254e159c3a01c125b847a9cad30. Not exactly critical, but we do still use `std::random_device urandom;` in `src/libaktualizr/utilities/utils.cc` for that and for `randomUuid()`.

Reply:
Ah, I see. Initially, it was added together with `std::rand()` for device names. After the change to `std:random_device`, `srand()` is not needed anymore.

Reply:
True, no reason to keep around the C functions when we have nice C++ ones, and it looks like you're right that nothin was actually using this anymore!